src/publisher.ts

```diff
@@ -179,3 +179,32 @@ export class Publisher {
     this.typesPublished[typeName] = true
   }
 }
+
+/**
+ * Get the composition-order pipeline of Nexus type def functions that will match the nullability and list types of the Prisma field type.
+ *
+ * For example { isList:true, isRequired: true } would result in array of funcs: [nonNull, list, nonNull]
+ */
+export const getNexusTypesCompositionForInput = (fieldType: DmmfTypes.SchemaArg['inputType']) => {
+  if (fieldType.isList) {
+    return [Nexus.list, Nexus.nonNull]
```

Comment:
In these functions the order you see here is as you would with `t.*` or `a(b(c()))`.

That is why above we `reduceRight`, rather than `reduce`.

This point is only actually applicable to this case so leaving the comment on this line.

(all other cases are only one type augmentation or symmetrical on both sides)